tests/components/dsmr/test_sensor.py

```diff
@@ -47,6 +49,40 @@ async def connection_factory(*args, **kwargs):
     return connection_factory, transport, protocol
 
 
+async def test_setup_platform(hass, mock_connection_factory):
+    """Test setup of platform."""
+    async_add_entities = MagicMock()
+
+    entry_data = {
+        "platform": DOMAIN,
+        "port": "/dev/ttyUSB0",
+        "dsmr_version": "2.2",
+        "precision": 4,
+        "reconnect_interval": 30,
+    }
+
+    with patch("homeassistant.components.dsmr.async_setup", return_value=True), patch(
+        "homeassistant.components.dsmr.async_setup_entry", return_value=True
+    ):
+        assert await async_setup_component(
+            hass, SENSOR_DOMAIN, {SENSOR_DOMAIN: entry_data}
+        )
+
+    assert not async_add_entities.called
+
+    await hass.async_block_till_done()
```

Comment:
youll wqnt this line as part of your with above to ensure the patch applies during duration of setup and import of config entry